tensorboard/plugins/base_plugin.py

```diff
@@ -32,16 +32,21 @@ class TBPlugin(object):
   """TensorBoard plugin interface. Every plugin must extend from this class.
 
   Subclasses must have a constructor that takes a TBContext argument.
+
+  Fields:
+    plugin_name: The plugin_name will also be a prefix in the http handlers 
+        generated by the plugin, e.g. `data/plugins/$PLUGIN_NAME/$HANDLER` The 
+        plugin name must be unique for each registered plugin, or a ValueError 
+        will be thrown when the application is constructed. The plugin name must 
+        only contain characters among [A-Za-z0-9_.-], and must be nonempty, 
+        or a ValueError will similarly be thrown.
+    disable_path_prefix: If True then data/plugins/$PLUGIN_NAME prefix will not
```

Comment:
I agree that we shouldn't have the `disable_path_prefix` field because I would like to (in a follow-up pull request) remove the functionality from application.py that inserts the `/data/plugin/FOO` prefix. We can talk more offline, but my rationale is that:

1. The user specifies true absolute paths on ts_web_library's path attribute.
2. The user specifies true absolute paths in his html files.
3. Plugins should specify true absolute paths too.
4. Prefix paths should be supported.
4. TensorBoardWSGI's role to check that no path collisions exist across plugins.